bentoml/deployment/sagemaker.py

```diff
@@ -0,0 +1,181 @@
+    identity = sts_client.get_caller_identity()
+    sts_arn = identity["Arn"]
+    sts_arn_list = sts_arn.split(':')
+    type_role = sts_arn_list[-1].split('/')
+    iam_client = boto3.client("iam")
+    if type_role[0] == 'user':
+        role_list = iam_client.list_roles()
+        arn = None
+        for role in role_list['Roles']:
+            policy_document = role['AssumeRolePolicyDocument']
+            statement = policy_document['Statement'][0]
+            if statement['Effect'] == 'Allow' and statement['Principal'][
+                    'Service'] == 'sagemaker.amazonaws.com':
+                arn = role['Arn']
+        if arn is None:
+            raise ValueError(
+                "Can't find proper Arn role for Sagemaker, please create one and try again")
+        return arn
+    elif type_role[0] == 'role':
+        role_response = iam_client.get_role(RoleName=type_role[1])
+        return role_response["Role"]["Arn"]
+
+
+def create_push_image_to_ecr(bento_service, snapshot_path):
+    """Create BentoService sagemaker image and push to AWS ECR
+
+    :param bento_service: Bento Service
+    :param snapshot_path: Path
+    :return: ecr_tag: String
+    """
+
+    # Example:
+    # https://github.com/awslabs/amazon-sagemaker-examples/blob/master/advanced_functionality/scikit_bring_your_own/container/build_and_push.sh
+    # 1. get aws account info and login ecr
+    # 2. create ecr repository, if not exist
+    # 3. build tag and push docker image
+
+    ecr_client = boto3.client('ecr')
+    token = ecr_client.get_authorization_token()
+    logger.info('Getting docker login info from AWS')
+    username, password = base64.b64decode(
+        token['authorizationData'][0]['authorizationToken']).decode('utf-8').split(":")
+    registry_url = token['authorizationData'][0]['proxyEndpoint']
+
+    # https://github.com/docker/docker-py/issues/2256
+    subprocess.call(['docker', 'login', '-u', username, '-p', password, registry_url])
+    docker_client = docker.from_env()
+    docker_client.login(username, password, email='', registry=registry_url)
+
+    image_name = bento_service.name.lower() + '-sagemaker'
+    ecr_tag = strip_scheme('{registry_url}/{image_name}:{version}'.format(
+        registry_url=registry_url, image_name=image_name, version=bento_service.version))
+
+    try:
+        ecr_client.describe_repositories(repositoryNames=[image_name])['repositories']
+    except ecr_client.exceptions.RepositoryNotFoundException:
+        ecr_client.create_repository(repositoryName=image_name)
+
+    print('Building docker image: {}'.format(image_name))
+    built_image = docker_client.images.build(path=snapshot_path, tag=image_name)
+    built_image[0].tag(ecr_tag)
+    print('Pushing image to AWS ECR at {}'.format(ecr_tag))
+    docker_client.api.push(repository=ecr_tag)
+    print('Finished pushing image: {}'.format(ecr_tag))
+    return ecr_tag
+
+
+def deploy_with_sagemaker(archive_path, additional_info):
```

Comment:
since you are also adding the status command, it might be good to check if there is already a deployment created and ask the user if they want to replace existing deployment. Make sure to add a `-y` option to skip the prompt.

Reply:
done, without the -y option, table that for later on